packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -17,6 +17,40 @@ import 'signals.dart';
 // the tool is killed by a signal.
 export 'package:file/file.dart';
 
+extension FileUtils on File {
+  /// Delete the file and return true if it exists, take no action and return
+  /// false if it does not.
+  ///
+  /// This method should be prefered to checking if it exists and
+  /// then deleting, because it handles the edge case where the file is
+  /// deleted by a different program between the two calls.
+  bool deleteIfExists({bool recursive = false}) {
+    if (!existsSync()) {
+      return false;
+    }
+    try {
+      deleteSync(recursive: recursive);
+    } on FileSystemException catch (err) {
```

Comment:
If `File` is an `ErrorHandlingFile`, then you could get either a `ToolExit` or an `Exception` here depending on whether you're under a `noExitOnFailure`.

Reply:
The goal with this catch is to only intercept error code 2, which we don't catch with error handling io because it normally seems to indicate a tool error.

Any other exception types should pass through as is, though I will add support for the existing error ignoring logic